### DISCUSSION TITLE
Prediksi menggunakan metode NB, DTree, Random Forest

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#118] [Q&A] [answered]
Dear Team Algoritma,

Mohon sarannya untuk project yang saya buat untuk membuat prediksi menggunakan metode Naive Bayes, DTree dan Random Forest. Object nya adalah untuk memprediksi apakah Kantor Cabang masuk dalam kategori Low/High dan masuk dalam jadwal audit atau tidak.

Kendala yang saya dapatkan adalah:

- Apakah untuk penentuan sampel, saya perlu melakukan upsampling atau tidak ? saya mencoba melakukan proses upsampling terlebih dahulu sebelum melakukan split data test dengan train. Namun ketika saya membatalkan proses upsampling, pada metode Random Forest terjadi error saat melakukan pada data test.
- Mohon reviu nya, apakah untuk proses analisisnya sudah tepat atau ada yg masih salah.

Mohon arahannya. Untuk dataset yang saya gunakan dan rmd nya terlampir pada link:
https://drive.google.com/file/d/1dXlgWuDLnJB_h5kf_4-mUneMcP9kBbM3/view?usp=sharing

Terima kasih.

**Comment by kevbow** [comment, marked as answer]
Dear @joejnt,

Saya belum berhasil membuka link yang Anda berikan. 
Perlu ditekankan bahwa proses upsampling atau downsampling hanya digunakan ketika kita bertemu data dengan variabel target imbalance, selain itu proses upsampling **hanya** digunakan pada training data. Untuk testing data kita biarkan saja seperti kondisi aslinya. Sedangkan untuk error belum dapat saya trace masalahnya.

Demikian informasi yang dapat saya sampaikan. Jika jawaban ini sesuai dengan permasalahan Anda mohon untuk memberikan tanda Answered pada jawaban ini, terima kasih.

**Reply by joejnt** [reply]
Dear @kevbow 

berikut adalah link untuk dataset
https://drive.google.com/file/d/1dXlgWuDLnJB_h5kf_4-mUneMcP9kBbM3/view?usp=sharing

berarti untuk proses upsampling dan down sampling dilakukan setelah melakukan splitting data ya ? mohon masukannya.

Thx u

**Reply by kevbow** [reply]
Jumlah data yang Anda gunakan sangatlah sedikit dan memang memiliki sifat yang imbalance sehingga lebih cocok untuk menggunakan upsampling. Anda dapat mencoba membandingkan model tanpa upsampling dengan model hasil upsampling. 
Untuk proses upsamplingnya dilakukan setelah splitting data hanya pada data train saja.

**Comment by joejnt** [comment]
@kevbow 

well noted, terima kasih untuk evaluasi dan saran nya.

Best regards,
JT
